routes/topics.js

```diff
@@ -5,12 +5,15 @@ const router = express.Router();
 // controller
 const topicsController = require('../controllers/topics');
 const {isAuth} = require('../middlewares/auth');
+const {validate} = require('../middlewares/joi-validation/validate');
+const {TopicValidation} = require('../joi-validations/topics.validations');
 const QueryParamsValidationMiddleware = require('../middlewares/query-params-validation');
 
 /* GET locations listing. */
 router.get('/list', topicsController.topics);
 router.get('/followed', isAuth, topicsController.getFollowedTopic);
 router.get('/follow', isAuth, topicsController.getFollowTopic);
+router.get('/latest', validate(TopicValidation.latestPost), isAuth, topicsController.getLatestPost);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The order of middlewares matters in Express.js. The `isAuth` middleware should be placed before the `validate` middleware. This way, we ensure that only authenticated users can trigger validation and reach the controller. 
```diff
- router.get('/latest', validate(TopicValidation.latestPost), isAuth, topicsController.getLatestPost);
+ router.get('/latest', isAuth, validate(TopicValidation.latestPost), topicsController.getLatestPost);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->